src/apps/loyalty/server/middlewares/user.ts

```diff
@@ -0,0 +1,21 @@
+import { NextFunction, Request, Response } from "express"
+import { fetchCollectorProfile } from "../gravity"
+
+export default (req: Request, res: Response, next: NextFunction) => {
```

Comment:
Yeah, I actually had this dilemma as well. I was going to use `res.locals` but ended up resorting to the user model since I had doubts whether properties would get cleared on res.locals after each request. I'm looking into it now.